### DISCUSSION TITLE
Output match file

### DISCUSSION THREAD

**Discussion by EZarnosky** [Fribb/MyAnimeList.bundle#44] [Ideas]
Would it be possible for the MAL Bundle to output a .match file to the directory of a matched series (kind of like the LAMDA bundle does for NFO and other files)? I have over 5000 series and my only option for match files would be programmatical.

If this is not possible maybe a separate tool to help by scanning all folders, suggesting matches and upon ok, output a .match file.

**Comment by Fribb** [comment]
Technically speaking, this should be possible. There shouldn't be much of a difference between the agent reading the `.match` file and writing one.

But I would still have some thoughts about this:

1. the `.match` file was implemented that way on purpose to allow other agents (if this should be the case) to use the same thing and not create other match files if they choose to have such a feature. (maybe it would be better to revert back to the `.malmatch` to be more in line with the `.plexmatch` file?!)
2. if the `.match` file is kept as-is, the agent should only change its own value, the mal-id.
3. this should also only be an "either-or" feature. So either you request metadata or you write metadata to the `.match` file. But not both

On the other hand, this feature would be used quite rarely. This might be better implemented as a standalone script that then interfaces with Plex or even a separate agent (like the already mentioned lambda) that does this.

A separate agent would also allow you to run them both in tandem through the Agent Priority list. So you, for example, run the MAL Agent first to search and save the metadata and then the other agent to save the `.match` file.

**Comment by EZarnosky** [comment]
I think a .malmatch would be great. I can't tell you how annoying it is to have a series I got 7 years ago and matched fine is now under a different name at mal and won't match anymore.

I agree after thinking about it, a standalone script might be better, maybe something with a GUI frontend?

I did a recent rebuild of my plex server after a major DB corruption and my anime library is all messed up. I need for force match a lot of series and fix several incorrect matches (5000 series and 1000 movies). a GUI tool (maybe in a spreadsheet format) that provides a best match and then allows me to correct it if need be would be very helpful.

I am sure whatever you might create would be of great use to everyone as the MAL bundle is too.

Any thoughts to a TinyMediaManager plug in? I know it does not create a match file, but it does grab everything else (posters, nfo, trailers, subs, etc.). Not sure if the plugin can be set to create a custom file like a .malmatch

**Comment by Fribb** [comment]
first things first, I converted the Issue to a Discussion because, well, this is more of a discussion that might result in a more precise issue rather than the issue itself. As of right now, I don't really see much for the agent to be changed for that.

> I think a .malmatch would be great. I can't tell you how annoying it is to have a series I got 7 years ago and matched fine is now under a different name at mal and won't match anymore.

technically speaking, you already have that through the `.match` file. I already use that in some cases in which ShokoServer uses some Title that isn't how MyAnimeList has them listed which would make matching impossible or hard to do resulting in possibly having to fix it manually.

But, yes, having to do that all over again for a large library will be a pain to do especially if something changed over the years.

As for a final solution, I would have to think about it. A dedicated script would be faster and GUI interaction would require quite a lot of time depending on how you do it and what you use. especially with the interaction between Plex and creating those files might be more work than it is worth.

As for TinyMediaManager, I took a quick look at it and it seems to be written in Java, so a language I know, but for this, you basically would have to recreate the Agent as a scraper to do the matching and then only save the ID in the file, since it is Java, this also should be possible and you could probably copy the same behaviour from the NFO file generation.

But, again, this is just a very surface-level look at things with a few assumptions.

Personally, I would rather prefer the Plex Metadata Agent way.

The biggest advantage of this is definitely that you don't really do much work on it, you would only need to take the ID and write it into a file of the show/movie you currently have. You could use it in tandem with both the MAL Agent and this Agent to "double team" it and automatically store the ID after a successful match.

However, this is also a disadvantage, at least for you because it won't help you get things up and running again quickly because you need to have a fixed library first and make sure that everything is matched correctly before being able to use it, otherwise the MALAgent would do what it already does by reading and force match your library items to the appropriate show.
